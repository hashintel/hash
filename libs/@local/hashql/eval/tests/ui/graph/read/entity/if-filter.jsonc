//@ run: fail
//@ description: Test if filter with branch fails
// biome-ignore format: readability
["::graph::tail::collect",
  ["::graph::body::filter",
    ["::graph::head::entities", ["::graph::tmp::decision_time_now"]],
    ["fn", { "#tuple": [] }, { "#struct": { "vertex": "_" } }, "_",
      ["if", {"#literal": true}, {"#literal": true}, {"#literal": false}]
      //~^ ERROR conditional expressions are not supported in filter contexts
    ]
  ]
]
