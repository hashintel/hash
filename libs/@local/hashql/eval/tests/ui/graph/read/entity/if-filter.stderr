error[graph-read-compiler::branch-unsupported]: Branch construct unsupported
  ╭▸ 
8 │       ["if", {"#literal": true}, {"#literal": true}, {"#literal": false}]
  │       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ conditional expressions are not supported in filter contexts
  │
  ├ help: rewrite the logic to avoid conditional statements
  ├ note: conditional statements (`if`/`else`) are not supported by the current query compiler. This is a fundamental limitation that will be addressed in the next-generation compiler
  ╰ note: consider using boolean operators (`&&`, `||`, `!`) to combine conditions directly instead of `if` statements